doc/tutorial/faq.txt

```diff
@@ -0,0 +1,68 @@
+.. _faq:
+
+===========================
+Frequently Asked Questions
+==================================
+How to update a subset of weights?
+==================================
+If you want to update only a subset of a weight matrix (such as
+some rows or some columns) that are used in the forward propogation
+of each iteration, then the cost function should be defined in a way
+that it only depends on the subset of weights that are used in that
+iteration.
+
+For example if you want to learn a lookup table, e.g. used for
+word embeddings, where each row is a vector of weights representing
+the embedding that the model has learned for a word, in each iteration,
+the only rows that should get updated are those containing embeddings
+used during the forward propagation. Here is how the theano function
+should be written:
+
+>>> # defining a shared variable for the lookup table
+>>> lookup_table = theano.shared(matrix_ndarray).
+>>>
```

Comment:
These empty lines are not correctly highlighted, I'm not sure how it should be fixed.